tensorflow/python/ops/string_ops.py

```diff
@@ -117,6 +118,7 @@ def string_split(source, delimiter=" "):  # pylint: disable=invalid-name
 ops.NotDifferentiable("ReduceJoin")
 ops.NotDifferentiable("StringJoin")
 ops.NotDifferentiable("StringSplit")
+ops.NotDifferentiable("Substr")
```

Comment:
Is this needed?  Typically you only need to add this if the op will ever be used in a context where its gradient might be asked for.  Given the use cases, I imagine this will never be backpropped on, and it would be better for this to raise an error due to 'no gradient function found' instead of silently truncating.